helmfile/values/dex.yaml.gotmpl

```diff
@@ -45,6 +45,13 @@ config:
       clientID: {{ .Values.dex.oktaClientID }}
       clientSecret: {{ .Values.dex.oktaClientSecret }}
       insecureSkipEmailVerified: true
+      insecureEnableGroups: true
```

Comment:
These two options are insecure so I think we should at least make it possible to disable them.

Reply:
insecureSkipEmailVerified we may have to leave as true in the okta set up because dex doesn't have access to that info from okta (as far as I can tell).  Will test the groups flag a bit more and see if we can secure that.  Notes from the dex docs on email verified:

 # Some providers return claims without "email_verified", when they had no usage of emails verification in enrollment process
    # or if they are acting as a proxy for another IDP etc AWS Cognito with an upstream SAML IDP
    # This can be overridden with the below option
    # insecureSkipEmailVerified: true 

Notes on insecureGroups:

 # Groups claims (like the rest of oidc claims through dex) only refresh when the id token is refreshed
    # meaning the regular refresh flow doesn't update the groups claim. As such by default the oidc connector
    # doesn't allow groups claims. If you are okay with having potentially stale group claims you can use
    # this option to enable groups claims through the oidc connector on a per-connector basis.
    # This can be overridden with the below option
    # insecureEnableGroups: true

Reply:
tested a bit more - without insecureEnableGroups, groups don't get passed by dex to appliaction (they are still sent by okta).  we could still make this an option but essentially if someone wants groups, we need to enable this option

Reply:
For the email_verified flag, here is the JWT token that okta sends to dex, note email_verified is not present:

{
  "sub": "00u1f7ceswDEKEnme5d6",
  "name": "Geoff Irwin",
  "email": "geoff.irwin@elastisys.com",
  "ver": 1,
  "iss": "https://elastisys-trial.okta.com/oauth2/default",
  "aud": "0oa1f9rmv0txIrlad5d6",
  "iat": 1607069958,
  "exp": 1607073558,
  "jti": "ID.6YzOn3h7wyN6JyU3RCn25USFUyljAnYbysuUyvoCiMg",
  "amr": [
    "pwd"
  ],
  "idp": "00o1c9issxiBuQqYV5d6",
  "nonce": "nonce",
  "preferred_username": "geoff.irwin@elastisys.com",
  "auth_time": 1000,
  "at_hash": "preview_at_hash",
  "groups": [
    "Everyone",
    "grafana_editor"
  ]
}

Reply:
Thanks for investigating! I think we want to make them configurable even with the limitations when they are disabled. There will always be someone who want to disable them for whatever reason. :slightly_smiling_face:

Reply:
cool, will do

Reply:
and done!